src/agents/coordinator.ts

```diff
@@ -95,16 +107,25 @@ export class DkgCoordinatorAgent {
     const Coordinator = await this.connectReadOnly(provider);
     // We leave `initiator` undefined because we don't care who the initiator is
     // We leave `successful` undefined because we don't care if the ritual was successful
-    const eventFilter = Coordinator.filters.EndRitual(
-      ritualId,
-      undefined,
-      undefined
-    );
-    Coordinator.once(eventFilter, (_ritualId, _initiator, successful) => {
+    const eventFilter = Coordinator.filters.EndRitual(ritualId, undefined);
+    Coordinator.once(eventFilter, (_ritualId, successful) => {
       callback(successful);
     });
   }
 
+  public static async getRitualIdFromPublicKey(
+    provider: ethers.providers.Provider,
+    dkgPublicKey: DkgPublicKey
+  ): Promise<number> {
+    const Coordinator = await this.connectReadOnly(provider);
+    const dkgPublicKeyBytes = dkgPublicKey.toBytes();
+    const pointStruct: BLS12381.G1PointStruct = {
+      word0: dkgPublicKeyBytes.slice(0, 16),
+      word1: dkgPublicKeyBytes.slice(16, 32),
```

Comment:
A G1Point is 48 bytes from the contract is defined as [follows](https://github.com/nucypher/nucypher-contracts/blob/main/contracts/contracts/lib/BLS12381.sol#L14).

Shouldn't `word0` by `(0,32)` and `word1` `(32,48)`?

Reply:
I think we need to start thinking about "acceptance" tests in `nucypher-ts` that interact with actual contracts (cc @cygnusv , @KPrasch ).

Reply:
IMHO acceptance tests with contracts are unnecessary. We're only supporting read operations (at this point), and setting those up is not different than existing mocks. Yes, this is a bug that went unnoticed. But it would have been caught easily by QA which we do on every release. 

I think the better solution to this particular problem is to enhance `BLS12381.G1PointStructOutput` generated type with a wrapper that performs validation. The `word0` and `word` fields are represented as strings by `typechain` in generated typings, which also prevented TS compiler from catching this issue.

Reply:
> But it would have been caught easily by QA which we do on every release.

The goal of the "acceptance" tests is to catch these things before live QA. It is inefficient to catch these kind of bugs during live QA on a testnet. Doesn't mean we will catch everything in those tests, but it does reduce the need for subsequent fixing/releases after bugs are found during live testing.

I don't know what specifically the "acceptance" tests would entail and what it would require from the codebase to be set up, but having everything tested with mocks limits the breadth of testing and bugs that we can find in tests - *who's testing the mocks* is essentially the argument.

"Acceptance" tests is a broad term, and maybe there is more appropriate term here, but we should investigate how we can perform tests closer to what the actual tesntet/production environment would be.

Reply:
I understand the advantages of acceptance testing, I just don't think they are necessary at this point. We should probably get them before the release, but not until we're done with monorepo or adapting `viem`: https://github.com/nucypher/nucypher-ts/pull/271

That being said, I'd be happy to accept a PR that implements acceptance testing if anyone wants to chip in.

Reply:
Definitely not something needed for this PR or this instant - but something we should nonetheless investigate. Filed #295 .